SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -229,6 +234,19 @@ class PositioningViewController: UIViewController, GMSMapViewDelegate, UITableVi
     }
     
     //MARK: Initializers
+    func addLocationListener() {
+        if let presenter = presenter {
+            presenter.addLocationListener()
+            changePositioningButton(toState: presenter.locationManager.state())
```

Comment:
Diria que so con cambiar o boton de posicion non é suficiente se o posicionamento estaba xa empezado, polo menos hai que amosar tamen a posicion do usuario no plano, cambiar o selector de plantas, etc
Para a parte de actualizar a pantalla crearia un metodo aparte e non o incluiría no addLocationListener, xa que creo que o codigo quedaría máis claro. 
Algo así como updateUI  ou metelo nese initializeUIElements, pero de primeira eu se engado un listener non dou por feito que se me actualiza a UI